packages/vx-tooltip/src/enhancers/withTooltip.tsx

```diff
@@ -1,28 +1,34 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 
-export const withTooltipPropTypes = {
-  tooltipOpen: PropTypes.bool,
-  tooltipLeft: PropTypes.number,
-  tooltipTop: PropTypes.number,
-  tooltipData: PropTypes.object,
-  updateTooltip: PropTypes.func,
-  showTooltip: PropTypes.func,
-  hideTooltip: PropTypes.func,
+export type WithTooltipProvidedProps = {
+  tooltipOpen?: boolean;
+  tooltipLeft?: number;
+  tooltipTop?: number;
+  tooltipData?: object;
+  updateTooltip?: (args: UpdateTooltipArgs) => void;
+  showTooltip?: (args: ShowTooltipArgs) => void;
+  hideTooltip?: () => void;
 };
+type WithTooltipState = Pick<
+  WithTooltipProvidedProps,
+  'tooltipOpen' | 'tooltipLeft' | 'tooltipTop' | 'tooltipData'
+>;
+type ShowTooltipArgs = Omit<WithTooltipState, 'tooltipOpen'>;
+type UpdateTooltipArgs = WithTooltipState;
+type WithTooltipContainerProps = { style: React.CSSProperties };
 
-export default function withTooltip(
-  BaseComponent,
-  containerProps = {
+export default function withTooltip<Props extends object = {}>(
```

Comment:
looks good, if the user specifies props on their component as `OwnProps & WithTooltipProvidedProps`, this should have everything on it 👍